tests/ops/models/test_pre_approval_webhook.py

```diff
@@ -0,0 +1,133 @@
+        )
+        connection_config = (
+            db.query(ConnectionConfig).filter_by(id=https_connection_config.id).first()
+        )
+        assert loaded_webhook.connection_config == connection_config
+        assert (
+            connection_config.pre_approval_webhooks
+            == all_webhooks_with_specific_connection_config
+        )
+
+        connection_config.delete(db=db)
+        for webhook in all_webhooks_with_specific_connection_config:
+            webhook.delete(db=db)
+
+    def test_create_connection_config_errors(
+        self, db: Session, https_connection_config, pre_approval_webhooks
+    ):
+        with pytest.raises(KeyValidationError) as exc:
+            new_webhook_no_key = PreApprovalWebhook.create(
+                db=db,
+                data={
+                    "connection_config_id": https_connection_config.id,
+                },
+            )
+            db.add(new_webhook_no_key)
+            db.commit()
+        assert str(exc.value) == "PreApprovalWebhook requires a name."
+
+        with pytest.raises(KeyOrNameAlreadyExists) as exc:
+            new_webhook_duplicate_key = PreApprovalWebhook.create(
+                db=db,
+                data={
+                    "connection_config_id": https_connection_config.id,
+                    "key": "pre_approval_webhook_2",
+                },
+            )
+            db.add(new_webhook_duplicate_key)
+            db.commit()
+        assert (
+            str(exc.value)
+            == "Key pre_approval_webhook_2 already exists in PreApprovalWebhook. Keys will be snake-cased names if not provided. "
+            "If you are seeing this error without providing a key, please provide a key or a different name."
+        )
+
+    def test_create_pre_approval_webhook_reply(
+        self,
+        db: Session,
+        privacy_request_status_pending,
+        https_connection_config,
+        pre_approval_webhooks,
+    ):
+        reply = PreApprovalWebhookReply.create(
+            db=db,
+            data={
+                "webhook_id": https_connection_config.id,
+                "privacy_request_id": privacy_request_status_pending.id,
+                "is_eligible": True,
+            },
+        )
+        db.add(reply)
+        db.commit()
+
+        loaded_reply = PreApprovalWebhookReply.get_by(
+            db, field="privacy_request_id", value=privacy_request_status_pending.id
+        )
+
+        assert loaded_reply.webhook_id == https_connection_config.id
+        assert loaded_reply.privacy_request_id == privacy_request_status_pending.id
+        assert loaded_reply.is_eligible is True
+        assert loaded_reply.created_at is not None
+
+        # assert relationship with PrivacyRequest
+        privacy_request = (
+            db.query(PrivacyRequest)
+            .filter_by(id=privacy_request_status_pending.id)
+            .first()
+        )
+        assert loaded_reply.privacy_request == privacy_request
```

Comment:
i like to also test the reverse, "privacy_request.pre_approval_webhook_reply".  There can be multiple replies right?  so shouldn't it be pre_approval_webhook_replies on the privacy request model side?

Reply:
that's a good idea and makes sense, as we can have multiple replies against the same privacy request 👍 

updating this now...